contracts/access/AccessControlDefaultAdminRules.sol

```diff
@@ -68,7 +70,21 @@ abstract contract AccessControlDefaultAdminRules is IAccessControlDefaultAdminRu
      * @inheritdoc IAccessControlDefaultAdminRules
      */
     function defaultAdminDelay() public view virtual returns (uint48) {
-        return _defaultAdminDelay;
+        return _curentDefaultAdminDelay;
```

Comment:
The "pending" delay becomes the "current" delay automatically (in `_beginDefaultAdminTransfer`).

This function returns the current delay even if the pending delay has already gone into effect. This presents a confusing view of the contract state, but most importantly it could mislead about the rules of the contract.

This pattern where some change is supposed to happen automatically is a little difficult because you can't encapsulate it in an internal function that gets the state and updates it, because you wouldn't be able to use it in a `view` context. But we should be able to find a clean way to implement it.

Reply:
We agreed to make the change automatic for the `defaultAdminDelay()` and the `pendingAdminDelay()`. I just implemented it in the last commit, however, [I added this warning.](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4079/files#diff-f3b8263b4d61197159b628b24b617c1b96139f713bdf96704e3cb30de436e4f0R32)

I think the admin's power to cancel the delay if it hasn't been written to `_currentDefaultAdminDelay` is non-trivial, what do you think?

Reply:
We decided to avoid another delay change to start or cancel the previous one if the schedule had already passed. In this way, it's guaranteed that the value will be kept when `defaultAdminDelay` is toggled.

Reply:
Implemented [here](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4079/commits/e99b5a41419aa6aa45b2d83ec957b586ee0c0bcd)